### ISSUE TITLE
weird character ESC[112 show up every time I run a command.

### ISSUE THREAD

**Issue by jdhao** [kingToolbox/WindTerm#188]
I am using windterm to connect to a remote server. I am using zsh for the remote server. Each time I run a command, I see weird character ESC]112 appear after the command.

**Comment by kingToolbox** [comment]
Sorry for the late reply. I just finished a long trip, and because there is no Internet, I can't deal with the issue in time. 

You are right. This is just because the control sequence `ESC]112` is not yet supported. WindTerm will support this in the next version 2.2.0. Thank you.

**Comment by kingToolbox** [comment]
The new [WindTerm_2.2.0_Prerelease_4](https://github.com/kingToolbox/WindTerm/releases/tag/2.2-prerelease) version has been released and `OSC 112` is supported, please download and check it, thank you.

**Comment by Nova38** [comment]
How do we enable it? I am trying to use windterm to connect to a serial port and it is not escaping the colors.  Here is the settings for the session 
![image](https://user-images.githubusercontent.com/17151865/140675883-fa6f123f-eeb5-457f-8c4b-9a81997f6e9e.png)

and here is what the serial output ends up looking like
![image](https://user-images.githubusercontent.com/17151865/140675924-f9016a02-ec4b-4981-aa0e-09e539feeaf6.png)

**Comment by kingToolbox** [comment]
There is no problem with the configuration, but from the screenshots, I don’t think the server outputs the correct `SGR control sequence` of colors. The correct `SGR control sequence` should `\033[m` or `\x1b[0m`, but there is no `ESC` in the screenshot. Can other clients like `PuTTY` display colors correctly?

If there is a problem, it may be that `ESC` is incorrectly parsed by `WindTerm`.

**Comment by Nova38** [comment]
![image](https://user-images.githubusercontent.com/17151865/140700643-0b22941e-6c63-410f-b66c-bb6c9f2ef401.png)

I have tired putty and mobaxterm and they both display the colors correctly.

**Comment by kingToolbox** [comment]
Yes, from the screenshot, `PuTTY` should indeed receive `ESC`. Can you collect the debug log for further analysis? If possible, the steps are as follows:

- Open the `Keyboard` session.
- Use the second button on the toolbar to disconnect the session.
- Check the menu item Menubar - Mode (7) - Debug Mode.
- Use the third button on the toolbar to reconnect the session.
- When the output is complete, uncheck the menu item Menubar - Mode (7) - Debug Mode.
- Click the menu item Right click menu - Log - Open Log Folder.
- Open the folder debug and paste your log file in your comment here. You can zip it before upload.

The log content is in plain text and will not contain any private information. You can use any editor to view it. Thank you.

**Comment by krobelus** [comment]
There seems ot be a bug in the [OSC parser](https://en.wikipedia.org/wiki/ANSI_escape_code#Operating_System_Command_sequences)
See https://github.com/fish-shell/fish-shell/issues/11609

Should be reproducible with with something like `printf '\x1b]123;foo\x07'`
